### DISCUSSION TITLE
Router can't properly return types with index signatures?

### DISCUSSION THREAD

**Discussion by SimHoZebs** [trpc/trpc#4584] [Q&A]
Below is the example code of my issue.

```ts
interface ITest {
  [key: string]: object | unknown;
  prop: string;
}

export const appRouter = router({
  foo: procedure.input(z.undefined()).query(() => {
    let test: ITest = { prop: "test" };

    return test;
  }),
});


export type AppRouter = typeof appRouter;

type bar = inferRouterOutputs<AppRouter>["foo"];
```

Here, `bar` is typed as
```ts
type bar = {
    [x: string]: never;
    [x: number]: never;
}
```
This results in a type mismatch, as the query data is no longer typed as `ITest` and has the `prop` property, contrary to the actual data. Is this default typescript behavior or trpc? How do I work around this problem?

Thanks in advance!

**Comment by microdev1** [comment]
Related:
- #5197
- #5211
